core/dbt/graph/graph.py

```diff
@@ -74,7 +74,22 @@ def get_subset_graph(self, selected: Iterable[UniqueId]) -> "Graph":
         new_graph = self.graph.copy()
         include_nodes = set(selected)
 
-        for node in self:
+        while True:
```

Comment:
The logic here looks correct, but we suggest a slightly different formulation which avoids using while True:

```python
        still_removing = True
        while still_removing:
            nodes_to_remove = list(
                node
                for node in new_graph
                if node not in include_nodes and new_graph.degree(node) < 2
            )
            if len(still_removing) == 0:
                still_removing = False
            else:
                new_graph.remove_nodes_from(nodes_to_remove)
```